ui/src/buckets/components/BucketList.tsx

```diff
@@ -26,7 +26,7 @@ import {SortTypes} from 'src/shared/utils/sort'
 // Utils
 import {prettyBuckets} from 'src/shared/utils/prettyBucket'
 
-type SortKey = keyof PrettyBucket
+type SortKey = keyof PrettyBucket | 'retentionRules[0].everySeconds'
```

Comment:
this is kind of setting off a spidey sense - seems like we're coupling the state and shape of our redux store to a value type - the end result is that the sorting code relies on the precise shape and existence of the store having a first item in `retentionRules` with an `everySeconds` value. if any of that isn't there or the shape changes (maybe the api gets updated), this could very likely break.

here's the thing though, i know you didn't make this pattern, and i don't know of a better way to do this without changing a lot of code, so i can't really do more than say keep en eye out on this sorting code cause it might break unexpectedly. (said without judgement of the quality of the code or the circumstances it was written under :))

Reply:
hopefully is things go south the tests will catch it, but yeah it is slightly concerning, but i dont think right now is the time to refactor all the logic. Well keep an eye to see if any related issues rear their head